packages/maskbook/src/extension/background-script/EthereumServices/types.d.ts

```diff
@@ -0,0 +1,10 @@
+import type { EventEmitter } from 'events'
+import type { AbstractProvider } from 'web3-core'
+
+export class MetamaskInpageProvider extends EventEmitter implements AbstractProvider {
```

Comment:
Did not Metamask inpage provider ship its own type?

Reply:
Its own Type `MetaMaskInpageProvider` is not accepted by `new Web3()`, so we should still keep this part:

```
Argument of type 'MetaMaskInpageProvider' is not assignable to parameter of type 'provider'.
```

The reason are some differences here, e.g.
https://github.com/MetaMask/inpage-provider/blob/master/index.d.ts#L48 https://github.com/ethereum/web3.js/blob/1.x/packages/web3-core/types/index.d.ts#L430